NExtends/Collections/DictionaryListMapper.cs

```diff
@@ -0,0 +1,234 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace NExtends.Collections
+{
+	/// <summary>
+	/// Helper class mapping entries of an IList&lt;<typeparamref name="V"/>&gt; to an IDictionary with configurable key values
+	/// </summary>
+	/// <example>
+	/// <code>
+	///		var test = new string[] { "entry1", "entry2" };
+	///
+	///		var entries = new DictionaryListMapEntry[] {
+	///			new  DictionaryListMapEntry(0, "lala"),
+	///			new  DictionaryListMapEntry(1, "lulu")
+	///		};
+	///		var m = new DictionaryListMapper&lt;string&gt;(test, entries);
+	///
+	///		Assert.IsTrue(m["lala"] == "entry1");
+	///		Assert.IsTrue(m["lulu"] == "entry2");
+	/// </code>
+	/// </example>
+	/// <typeparam name="V"></typeparam>
+	public class DictionaryListMapper<V> : IDictionary<string, V>
+	{
+		IList<V> CollectionToMap;
+		IEnumerable<IDictionaryListMapEntry> MapEntries;
+		ICollection<string> MapEntriesKeys;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the 'MapEntriesKeys' field and declare it as a local variable in the relevant methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=csharpsquid%3AS1450)